### PR TITLE
KAA-1247: [C/C++ SDK] Rename x86 platform

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @forGGe 
Requires https://github.com/kaaproject/kaa/pull/866

### CODE REVIEW

**Comment by forGGe** [comment]
Looks good

File: common/c/targets/posix/target.h [hunk 1]
```diff
@@ -15,22 +15,21 @@
  */
 
 /*
- * This header provides several bindings for x86_64 target that abstracts
+ * This header provides several bindings for posx target that abstracts
```
Comment (rasendubi) [inline common/c/targets/posix/target.h]: posix

File: common/c/targets/posix/target.h [hunk 2]
```diff
@@ -15,22 +15,21 @@
...
  * an implementation of its features. Right now it contains only
  * console and target initialisation routines, but it must be extended
  * if required.
  *
- * Note that by x86 target is understood a system with support of
- * stdc library and most likely running full-featured OS.
+ * Note that posix target does imply a system with support of
```
Comment (rasendubi) [inline common/c/targets/posix/target.h]: implies

File: configurationdemo/source/c/build.sh [hunk 3]
```diff
@@ -46,22 +46,18 @@ KAA_PRODUCE_BINARY=
...
-    x86-64)
-        KAA_TARGET=x86-64
+    posix)
+        KAA_TARGET=posix
         ;;
 
     "")
-        # Do nothing if string is empty - x86 is requested
-        KAA_TARGET=x86-64
+        # Do nothing if string is empty - posix is requested
```
Comment (rasendubi) [inline configurationdemo/source/c/build.sh]: `KAA_TARGET=posix` is not "nothing"

File: configurationdemo/source/cpp/build.sh [hunk 4]
```diff
@@ -20,7 +20,7 @@ RUN_DIR=`pwd`
 
 function help {
     echo "Choose one of the following: {build|run|deploy|clean}"
-    echo "Supported platforms: x86-64, edison"
+    echo "Supported platforms: posix, edison"
```
Comment (rasendubi) [inline configurationdemo/source/cpp/build.sh:23]: remove edison?

Comment (rasendubi) [inline configurationdemo/source/cpp/build.sh:23]: or return it back above

Comment (forGGe) [inline configurationdemo/source/cpp/build.sh:23]: Edison cross-compilation is supported by the C++ SDK, thus should be kept here

**Pull request merged**